src/pages/login.tsx

```diff
@@ -17,14 +17,22 @@ export const getServerSideProps : GetServerSideProps = async () => {
     const protocol = stringToBool(process.env.NEXT_PUBLIC_APP_USE_TLS)? 'https' : 'http';
     const host = process.env.NEXT_PUBLIC_APP_HOST;
 
+    let scopes;
```

Comment:
why is Typescript letting you declare a variable that doesn't know what it will get?

Reply:
I guess TypeScript is able to infer the type from what happens to the variable further down in the function. Judging by this screenshot from VSCode (that knows TypeScript quite well) that seems to be what's happening.

There are two (relevant) branches of code below, one in which `scopes` does not get a value at all (i.e. `undefined`) and one where it becomes a string array. So the type is `undefined | string[]`.

![image](https://user-images.githubusercontent.com/550212/117318774-060af280-ae8b-11eb-90c0-7a5385985161.png)